Enable removal of UNIX lock files

sbc100 opened the pull request:
This avoids leaving orphan locks files around after we are done when them.

See https://github.com/emscripten-core/emscripten/pull/24035

Fixes: #31

gaborbernat reviewed (changes requested):
Please add tests 👍

sbc100 commented on src/filelock/_unix.py:
```diff
@@ -38,9 +38,7 @@ class UnixFileLock(BaseFileLock):
-            open_flags = os.O_RDWR | os.O_TRUNC
-            if not Path(self.lock_file).exists():
-                open_flags |= os.O_CREAT
+            open_flags = os.O_RDWR | os.O_TRUNC | os.O_CREAT
```
I worry a little that I had to remove this workaround which was added in: #317.

I can't seem to find a way to keep both of these fixes in place.

gaborbernat replied:
Yeah, I'm hesitant to approve it without it though 🤔

gaborbernat committed:
✨ feat(unix): delete lock file on release

Lock files are now cleaned up when released, preventing orphan .lock
files from accumulating. The acquire/release cycle handles three
race conditions:

- Sticky-bit directories (#317): O_CREAT on a file owned by another
  user raises PermissionError, so we fall back to opening without
  O_CREAT when the file already exists.
- Deleted-before-open: if the file vanishes between exists() and
  os.open(), the caller's retry loop recreates it.
- Deleted-before-flock: if another process unlinks the file between
  our open() and flock(), st_nlink==0 detects the stale inode and
  the retry loop starts fresh.

Fixes #31.
